src/Spice86.Core/Emulator/CPU/CPU.cs

```diff
@@ -85,6 +87,7 @@ public class Cpu {
 
     public void ExecuteNextInstruction() {
         _internalIp = State.IP;
+        RememberState(State.ToString());
```

Comment:
What is the impact on perfs for this?
I think we could implement it another way, at some point spice86 will have to maintain a trace of all that has been executed in a structured form. Last executed path could be traced from there.

Reply:
This is debugging info that I use and was not meant to be committed to this branch :(
I have now reverted those changes.